inference-engine/src/gna_plugin/gna_graph_tools.hpp

```diff
@@ -711,4 +711,22 @@ inline void CNNNetworkRemoveLayer(CNNLayerPtr layer, bool checkDims = true) {
 
     // removing layer->osp, and layer->isp connection not necessary - layer will delete it by itself
 }
+
+/**
+ * @brief returns dimensions order for permute from input to output layout
+ * @param in_layout
+ * @param out_layout
+ */
+inline std::vector<int> GetPermuteOrder(Layout in_layout, Layout out_layout) {
+    if (in_layout == out_layout) {
+        return {0, 1, 2, 3};
+    }
+    if (in_layout == Layout::NHWC && out_layout == Layout::NCHW) {
+        return {0, 3, 1, 2};
+    }
+    if (in_layout == Layout::NCHW && out_layout == Layout::NHWC) {
+        return {0, 2, 3, 1};
+    }
+    THROW_IE_EXCEPTION << "Unsupported layouts " << in_layout << " and/or " << out_layout << " for permute";
```

Comment:
unfortunately it is hard to guess layer name for unsupported layout, recommend to move this message in outer code,  and right no more like a static assert according to usage of this function.